compatibility_server/pip_checker.py

```diff
@@ -27,29 +27,48 @@
 import enum
 import json
 import logging
-import os.path
 import shlex
-import subprocess
-import tempfile
+import socket
 import urllib.request
 
 from typing import Any, List, Mapping, Optional
 
+import docker
+
+socket.setdefaulttimeout(3)
```

Comment:
This is a bit of a code smell since it affects socket timeouts outside of the module. Since this server is the only place that this is used it will work but it might be better to use explicit timeouts when fetching URLs, if possible.

Reply:
Removed. Timeout is also handled in the internal docker.